File: 04_bubble_sort/bubble_sort.py
from numpy import random

# generate
data = random.randint(99, size=(10))

# optimized bubble-sort
for j in range(len(data) - 1):
    swap = False
    for i in range(len(data) - 1 - j):
        if data[i] > data[i + 1]:
            data[i], data[i + 1] = data[i + 1], data[i]
            swap = True
    if not swap:
        break

# show
print(*data, sep=" ")

File: 04_bubble_sort/bubble_sort.js
// generate
var data = Array.from({ length: 10 }, () => Math.floor(Math.random() * 100));

// optimized bubble-sort
var tmp;
var swap;
for (let j = 0; j < data.length - 1; j++) {
  swap = false;
  for (let i = 0; i < data.length - 1 - j; i++) {
    if (data[i] > data[i + 1]) {
      tmp = data[i];
      data[i] = data[i + 1];
      data[i + 1] = tmp;
      swap = true;
    }
  }
  if (!swap) {
    break;
  }
}

// show
console.log(data.join(" "));
